### PR TITLE
TST: add the ability to pass kwargs through standard load tests

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #583

Modifies the instrument identification so that instruments may pass through a series of optional kwargs for `test_load`. The optional kwargs may be passed through as a dict or list of dicts describing each configuration.  Note that this only verifies that the data can be successfully loaded under this configuration, custom tests are required for testing specific outcomes.

Backwards-compatible with develop.  A new pytest mark is added (load_options).  Note that if this is unregistered in setup.cfg of downstream packages, the tests will still run and a warning will be generated asking the user to register the new mark.

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Testing with local tests of pysat/pysatNASA#91. 

```
pytest -vs pysat/tests/test_instruments.py 
```

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.11

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/tests/classes/cls_instrument_library.py [hunk 1]
```diff
@@ -36,11 +36,12 @@ def initialize_test_inst_and_date(inst_dict):
 
     """
 
+    kwargs = {} if 'kwargs' not in inst_dict.keys() else inst_dict['kwargs']
```
Comment (jklenzing) [inline pysat/tests/classes/cls_instrument_library.py:39]:
```suggestion
    kwargs = inst_dict['kwargs'] if 'kwargs' in inst_dict.keys() else {}
```

File: setup.cfg [hunk 2]
```diff
@@ -69,5 +69,6 @@ markers =
     all_inst: tests all instruments
     download: tests for downloadable instruments
     no_download: tests for instruments without download support
+    load_options: tests for instruments with optional load kwargs
```
Comment (jklenzing) [inline setup.cfg:72]:
```suggestion
    load_options: tests for instruments including optional load kwargs
```

**Comment by aburrell** [comment]
@jklenzing I was waiting on the check to give a review, but it looks like they're having trouble running.  May be related to the updates?

**Comment by jklenzing** [comment]
> @jklenzing I was waiting on the check to give a review, but it looks like they're having trouble running. May be related to the updates?

Looks like it just never ran this morning.  Just pushed a minor update to the changelog description, and they're running now.  The only changes to the code since the last successful run of the tests this morning have been to documentation, so it's probably fine to look over the code parts.

**Comment by aburrell** [comment]
@jklenzing I did look over the files and didn't find anything, but I'm not going to approve until the CI passes 🔭  🐞

**Comment by jklenzing** [comment]
Is the suggested syntax clear?  Went back and forth a number of times on this.

**Comment by aburrell** [comment]
It looked fine to me.

**Comment by jklenzing** [comment]
Woohoo!  The checks have run successfully.

**Pull request merged**